code/parse/sexp/sexp_lookup.cpp

```diff
@@ -217,6 +187,17 @@ DynamicSEXP* get_dynamic_sexp(int operator_const)
 
 	return iter->second.get();
 }
+int get_category_of_subcategory(int subcategory_id)
+{
+	auto& global = globals();
+
+	auto iter = global.subcategory_to_category.find(subcategory_id);
+	if (iter == global.subcategory_to_category.end()) {
+		return OP_CATEGORY_NONE;
+	}
```

Comment:
Right now this is fine, but if someone will use this method in the future, they'll be confused as to why this won't resolve built-in categories, as they're never added to the map.